bftengine/tests/testSeqNumForClientRequest/seqNumForClientRequest_test.cpp

```diff
@@ -1,8 +1,80 @@
 #include "gtest/gtest.h"
+#include "SimpleClientImp.cpp"
 
-TEST(seqNumForClientRequest_test, initial_test) {}
+using namespace std;
+using namespace bftEngine;
 
-int main(int argc, char* argv[]) {
+namespace {
+
+class mockSeqNumGenerator : public SeqNumberGeneratorForClientRequestsImp {
+ public:
+  uint32_t getLastCountOfUniqueFetchID() { return lastCountOfUniqueFetchID_; }
+
+  void setLastCountOfUniqueFetchID(uint64_t count) { lastCountOfUniqueFetchID_ = count; }
+};
+
+TEST(seqNumForClientRequest_test, send_request_fast_still_persistence) {
+  SeqNumberGeneratorForClientRequests *seqNumGen = new mockSeqNumGenerator();
```

Comment:
Why not `auto* seqNumGen = new mockSeqNumGenerator();`? This way you would probably not need to do dynamic casting several lines below...

Reply:
The function that I am testing declared at 'SeqNumberGeneratorForClientRequestsImp' as private, so I can't use it that way.